File: week2_algorithic_warm-up/gcd/gcd_python/gcd.py
from sys import stdin

def gcd(a, b):
    while True:
        remainder = a % b
        if (remainder == 0):
            return b
        else:
            a, b = b, remainder


def gcd_fast(a, b):
    if (b == 0):
        return a

    a_prime = a % b;
    return gcd_fast(b, a_prime)

if __name__ == '__main__':
    input = stdin.readline()
    a, b = map(int, input.split())
    print("{}".format(gcd(a, b)))  

File: week2_algorithic_warm-up/gcd/gcd_cpp/gcd.cpp
#include <iostream>


int GCD(int, int);

int main()
{
    int a, b;
    std::cin >> a >> b;
    std::cout << GCD(a, b) << std::endl;
    return 0;
}

int GCD(int a, int b) {
    if (b == 0)
        return a;
    int a_prime = a % b;
    return GCD(b, a_prime);
}
